monai/apps/deepgrow/transforms.py

```diff
@@ -652,8 +644,12 @@ def bounding_box(self, points, img_shape):
 
     def __call__(self, data):
         d: Dict = dict(data)
+        exist_keys = list(self.key_iterator(d))
```

Comment:
in this way we lose the nice iterator style... but at the same time I don't have a better idea to check whether the iterator has no item cc @rijobro @ericspod

Reply:
Hi @wyli @ericspod @rijobro ,

Thanks for your review here.
I think it's OK to get the length by listing the iterator as the `self.keys` are very small data, do you have any other ideas?

Thanks in advance.

Reply:
Hi @rijobro @ericspod ,

Could you please help share some comments here? Or any other better idea？

Thanks in advance.

Reply:
in this case perhaps we can have
```py
first_key = next(iterator, [])
if first_key == []:
    print("iterator empty")
```

Reply:
Sure, I updated the PR according to your suggestion.
Could you please help review it again?

Thanks in advance.

Reply:
Using monai.utils.first(iterator, []) will also improve the readability. Should we consider 'self.first_key(data)' a base class API?

Reply:
Actually, I planned to add to the base class API, but we usually don't change base class frequently, so I didn't do it.
Lst me add it to base class directly.

Thanks.